src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateClones = [];
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (stateClone.hasOwnProperty(key)) {
+            delete stateClone[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(stateClone)) {
```

Comment:
You can use [for in ](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/for...in)loop